### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by lrennels**
This PR updates all primary documents in the  _Mimi/docs/src folder_.  @rjplevin we will want to discuss how to finish the *internals* documents between us, if you want to take a first stab at completing them and then I can read through and comment for clarification ... or we can work on them together.

### CODE REVIEW

**Comment by lrennels** [comment]
@davidanthoff @rjplevin this is ready for review, it does not include updates to the internals documentation, as described above

File: docs/src/faq.md [hunk 1]
```diff
@@ -37,22 +37,22 @@ Here, `../data/cubicparams.csv` is a parameter definition file that looks someth
 
 ## How do I use component references?
 
-Component references allow you to write cleaner model code when connecting components.  The `addcomponent` function returns a reference to the component that you just added:
+Component references allow you to write cleaner model code when connecting components.  The `component` function returns a reference to the component that you just added:
```
[the remark this answers is not in the document]
Comment (lrennels) [reply docs/src/faq.md]: You're right, I was getting mixed up with the syntax of the @defmodel macro.

File: docs/src/tutorial.md [hunk 2]
```diff
@@ -4,110 +4,131 @@
...
-addcomponent(my_model, grosseconomy)  #Order matters here. If the emissions component were defined first, the model would not run.
-addcomponent(my_model, emissions)
+export m
+
+m = Model()
+
+set_dimension!(m, :time, collect(2015:5:2110))
+
+#Order matters here. If the emissions component were defined first, the model would not run.
+addcomponent(my_model, grosseconomy, :grosseconomy)
```
[the remark this answers is not in the document]
Comment (lrennels) [reply docs/src/tutorial.md]: Yep, that simpler syntax works I'll use that one.

File: src/core/build.jl [hunk 3]
```diff
@@ -87,7 +87,10 @@ function instantiate_component(md::ModelDef, comp_def::ComponentDef)
 
     # println("instantiate_component:\n  vtype: $vars_type\n\n  ptype: $pars_type\n\n  vvals: $var_vals\n\n  pvals: $par_vals\n\n")
 
-    comp_inst = ComponentInstance(comp_def, vars_type(var_vals), pars_type(par_vals), comp_name)
+    v = vars_type(var_vals)
+    p = pars_type(par_vals)
+
+    comp_inst = ComponentInstance{typeof(v), typeof(p)}(comp_def, v, p, comp_name)
```
[the remark this answers is not in the document]
Comment (rjplevin) [reply src/core/build.jl]: This was a change I made at some point. I wonder if Lisa's changes were against master or the branch I had been working on.

[the remark this answers is not in the document]
Comment (lrennels) [reply src/core/build.jl]: It was probably something like this, I think it's fixed now.

**Pull request merged**